src/System/extract_system_rep.rs

```diff
@@ -149,6 +149,19 @@ impl SystemRecipe {
             },
         }
     }
+
+    /// Takes in an output [Vec]<[Component]> and pushes only the components to the vector
+    pub fn get_components(&self, out: &mut Vec<Component>) {
+        match self {
+            SystemRecipe::Composition(left, right)
+            | SystemRecipe::Conjunction(left, right)
+            | SystemRecipe::Quotient(left, right, _) => {
+                left.get_components(out);
+                right.get_components(out)
+            }
+            SystemRecipe::Component(comp) => out.push(*comp.to_owned()),
+        }
+    }
```

Comment:
Change this code with the following implementation. This implementation ensures that `get_components` returns a `Vec` and requires no `Vec` as input parameter
```
pub fn get_components(&self) -> Vec<Component>{
      let mut components: Vec<Component> = Vec::new();
      self.get_components_helper(&mut components);
      components
}

fn get_components_helper(&self, out: &mut Vec<Component>) {
    match self {
        SystemRecipe::Composition(left, right)
        | SystemRecipe::Conjunction(left, right)
        | SystemRecipe::Quotient(left, right, _) => {
            left.get_components_helper(out);
            right.get_components_helper(out)
        }
        SystemRecipe::Component(comp) => out.push(*comp.to_owned()),
    }
}
```